python/taichi/lang/matrix.py

```diff
@@ -482,7 +476,7 @@ def __init__(self,
             assert ndim in (0, 1, 2)
             self.ndim = ndim
 
-        if self.n * self.m > 32 and not suppress_warning:
+        if self.n * self.m > 32 and is_logging_effective('warn'):
```

Comment:
IIRC `is_logging_effective` is more about our C++ logging but `warning` below isn't controlled by that. Maybe it makes more sense to think about how to make sure `ti.lang.util.warning` respect our log_level set in `ti.init`? 
In the worst case we can replace the `warning` below with `ti.warn` from C++?

Reply:
I have test the code liek this:
```python
import taichi as ti
ti.init(log_level=ti.ERROR)


@ti.kernel
def make():
    v = ti.Vector([0] * 50)
    print(v[0])

make()
```

The warning msg will be printed if the logging level is set to `ti.ERROR` or `ti.CRITICAL`, and will be hidden is set to `ti.TRACE`. I think this is our desired purpose?

Reply:
Yep it's the desired behavior but spreading `is_logging_effective` checks over python codebase isn't a clean way to do this ;)

Reply:
That makes sense. I have moved the check to the `warn` function, but I'm afraid this may cause unexpected effects, would you please have a look at this?